### PR TITLE
Modus Table: Update checkbox size based on density

### PR DESCRIPTION
**Pull request by yohernandez**
## Description

Adding a new property to rowSelectionOptions to handle checkbox size.

References #1964

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

Added e2e test to check if a small checkbox is rendered.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-table/models/modus-table.models.ts [hunk 1]
```diff
@@ -121,6 +121,7 @@ export interface ModusTableRowSelectionOptions {
   multiple?: boolean;
   subRowSelection?: boolean;
   preSelectedRows?: string[];
+  checkboxSize: 'small' | 'medium';
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/models/modus-table.models.ts:124]: Adding a new property to control the checkbox's size.

File: stencil-workspace/src/components/modus-table/modus-table.e2e.ts [hunk 2]
```diff
@@ -894,4 +894,27 @@ describe('modus-table', () => {
     await page.waitForChanges();
     expect(element).toHaveClass('density-compact');
   });
+
+  it('Renders small size checkboxes', async () => {
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/modus-table.e2e.ts:898]: This test is checking that the checkbox's size is small when the `checkboxSize` property is set to 'small'

File: stencil-workspace/src/components/modus-table/modus-table.scss [hunk 3]
```diff
@@ -48,6 +48,10 @@ table {
     &.row-checkbox {
       padding: 0.5rem !important;
       width: 46px;
+
+      &.small {
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/modus-table.scss:52]: set the column width to 38px when the checkbox's size is small

File: stencil-workspace/src/components/modus-table/modus-table.scss [hunk 4]
```diff
@@ -358,7 +362,8 @@ table {
   }
 
   &.density-comfortable {
-    tbody {
+    tbody,
+    tfoot {
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/modus-table.scss:366]: this is fixing https://github.com/trimble-oss/modus-web-components/issues/1969

File: stencil-workspace/src/components/modus-table/modus-table.scss [hunk 5]
```diff
@@ -372,7 +377,8 @@ table {
   }
 
   &.density-compact {
-    tbody {
+    tbody,
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/modus-table.scss:380]: this is fixing https://github.com/trimble-oss/modus-web-components/issues/1969

File: stencil-workspace/src/components/modus-table/modus-table.tsx [hunk 6]
```diff
@@ -200,6 +200,7 @@ export class ModusTable {
   @Prop() rowSelectionOptions: ModusTableRowSelectionOptions = {
     multiple: false,
     subRowSelection: false,
+    checkboxSize: 'medium',
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/modus-table.tsx:203]: setting default value to medium

File: stencil-workspace/src/components/modus-table/modus-table.tsx [hunk 7]
```diff
@@ -208,7 +209,8 @@ export class ModusTable {
     if (
       newVal.multiple !== oldVal.multiple ||
       newVal.subRowSelection !== oldVal.subRowSelection ||
-      newVal.preSelectedRows !== oldVal.preSelectedRows
+      newVal.preSelectedRows !== oldVal.preSelectedRows ||
+      newVal.checkboxSize !== oldVal.checkboxSize
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/modus-table.tsx:213]: react to a change on the new property

File: stencil-workspace/src/components/modus-table/parts/modus-table-body.tsx [hunk 8]
```diff
@@ -60,7 +61,8 @@ export const ModusTableBody: FunctionalComponent<ModusTableBodyProps> = ({ conte
               <ModusTableCellCheckbox
                 multipleRowSelection={multipleRowSelection}
                 row={row}
-                isChecked={isChecked}></ModusTableCellCheckbox>
+                isChecked={isChecked}
+                checkboxSize={checkboxSize}></ModusTableCellCheckbox>
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/modus-table-body.tsx:65]: propagate the checkboxSize property to ModusTableCellCheckbox component

File: stencil-workspace/src/components/modus-table/parts/modus-table-footer.tsx [hunk 9]
```diff
@@ -35,7 +36,10 @@ export const ModusTableFooter: FunctionalComponent<ModusTableSummaryRowProps> =
       {getFooterGroups().map((group) => (
         <tr class="summary-row">
           {rowSelection && (
-            <td id={COLUMN_DEF_ROW_SELECTION_ID} key={COLUMN_DEF_ROW_SELECTION_ID} class={COLUMN_DEF_ROW_SELECTION_CSS}></td>
+            <td
+              id={COLUMN_DEF_ROW_SELECTION_ID}
+              key={COLUMN_DEF_ROW_SELECTION_ID}
+              class={COLUMN_DEF_ROW_SELECTION_CSS + ' ' + (rowSelectionOptions?.checkboxSize ?? '')}></td>
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/modus-table-footer.tsx:42]: adding the classes small or medium to the table td for the footer.

File: stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-cell-checkbox.tsx [hunk 10]
```diff
@@ -33,7 +35,7 @@ export const ModusTableCellCheckbox: FunctionalComponent<ModusTableCellCheckboxP
   }
   return (
     <td
-      class="row-checkbox sticky-left"
+      class={'row-checkbox sticky-left ' + (checkboxSize ?? '')}
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-cell-checkbox.tsx:38]: adding the classes small or medium to the table td for the body.

File: stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-cell-checkbox.tsx [hunk 11]
```diff
@@ -43,6 +45,7 @@ export const ModusTableCellCheckbox: FunctionalComponent<ModusTableCellCheckboxP
         ref={(el) => (checkboxInput = el)}
         checked={isChecked}
         indeterminate={multipleRowSelection && row.getIsSomeSelected()}
+        size={checkboxSize}
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-cell-checkbox.tsx:48]: set checkbox size property

File: stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx [hunk 12]
```diff
@@ -14,12 +14,13 @@ export const ModusTableHeaderCheckbox: FunctionalComponent<ModusTableHeaderCheck
...
   } = context;
   return (
-    <th class="row-checkbox sticky-left">
+    <th class={'row-checkbox sticky-left ' + (rowSelectionOptions?.checkboxSize ?? '')}>
       {rowSelectionOptions?.multiple && (
         <modus-checkbox
           ariaLabel="Select all rows"
           checked={getIsAllRowsSelected()}
           indeterminate={getIsSomeRowsSelected()}
+          size={rowSelectionOptions?.checkboxSize}
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx:23]: set the checkbox size property in the header

File: stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx [hunk 13]
```diff
@@ -14,12 +14,13 @@ export const ModusTableHeaderCheckbox: FunctionalComponent<ModusTableHeaderCheck
     rowSelectionOptions,
   } = context;
   return (
-    <th class="row-checkbox sticky-left">
+    <th class={'row-checkbox sticky-left ' + (rowSelectionOptions?.checkboxSize ?? '')}>
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx:17]: adding the classes small or medium to the table td for the header.

File: stencil-workspace/storybook/stories/components/modus-table/modus-table-storybook-docs.mdx [hunk 14]
```diff
@@ -481,6 +481,7 @@ To select rows, users can use the checkbox provided on each row. Normally, the M
   - To enable multiple row selection, set `multiple` to `true`.
   - Sub-rows will also be selected if the parent is selected by setting `subRowSelection` to `true`.
   - To load the table with pre selected rows, set preSelectedRows to a list of row ids. **Row ids default to the row index, but can also be set by passing an id value in data.**
+  - To use a different size for the checkboxes, set `checkboxSize` to `small`. Default: `medium`.
```
Comment (yohernandez) [inline stencil-workspace/storybook/stories/components/modus-table/modus-table-storybook-docs.mdx:484]: updating Storybook

**Comment by rthanga1** [comment]
We have to apply the same fix to the header row as well.

File: stencil-workspace/src/components/modus-table/modus-table.scss (hunk 4 above)
Comment (cjwinsor) [reply stencil-workspace/src/components/modus-table/modus-table.scss:366]: @yohernandez After discussing with Modus team, we don't see any reason why the header shouldn't honor the density settings. If you want to include that issue fix in this PR, you'll need to address the header.  Feel free to point me to your UX if there is a disagreement.

File: stencil-workspace/src/components/modus-table/models/modus-table.models.ts (hunk 1 above)
Comment (cjwinsor) [reply stencil-workspace/src/components/modus-table/models/modus-table.models.ts:124]: @yohernandez I am going to ask a few resources, but the general thought so far with Modus is that the user shouldn't have control of the textbox size.  If we are to make it a standard that we use the small checkbox for the compact density, then that shouldn't be configurable.  Can you ping me so we can discuss?

File: stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx (hunk 12 above)
Comment (cjwinsor) [reply stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx:23]: This is also why the header density came up, as the small checkbox looks out of place with the standard header size.

**Review by cjwinsor — CHANGES_REQUESTED** [review]
@yohernandez Can we discuss the need for the checkbox to switch to the small size?

File: stencil-workspace/src/components/modus-table/modus-table.scss (hunk 4 above)
Comment (yohernandez) [reply stencil-workspace/src/components/modus-table/modus-table.scss:366]: We had a first design where the density affects the body and footer only. We are double-checking. In the mean time, we'll pull out this change from this PR. Thanks.

**Comment by yohernandez** [comment]
@cjwinsor We'll go with your suggestion: "we use the small checkbox for the compact density". We'll update this PR properly.

File: stencil-workspace/src/components/modus-table/modus-table.e2e.ts [hunk 15]
```diff
@@ -894,4 +894,28 @@ describe('modus-table', () => {
     await page.waitForChanges();
     expect(element).toHaveClass('density-compact');
   });
+
+  it('Renders small size checkboxes for compact density', async () => {
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/modus-table.e2e.ts:898]: This test changes the table density to compact and verifies that the checkbox's size is small.

File: stencil-workspace/src/components/modus-table/parts/modus-table-body.tsx [hunk 16]
```diff
@@ -13,9 +13,13 @@ interface ModusTableBodyProps {
 }
 
 export const ModusTableBody: FunctionalComponent<ModusTableBodyProps> = ({ context }) => {
-  const { hover, rowSelection, rowSelectionOptions, rowActions, tableInstance: table, updateData } = context;
+  const { density, hover, rowSelection, rowSelectionOptions, rowActions, tableInstance: table, updateData } = context;
   const hasRowActions = rowActions?.length > 0;
   const multipleRowSelection = rowSelectionOptions?.multiple;
+  let checkboxSize: 'medium' | 'small' = 'medium';
+  if(density === 'compact') {
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/modus-table-body.tsx:20]: change the checkbox's size to small for compact density

File: stencil-workspace/storybook/stories/components/modus-table/modus-table-storybook-docs.mdx [hunk 17]
```diff
@@ -604,6 +604,7 @@ interface ModusTableCellValueChange {
 interface ModusTableRowSelectionOptions {
   multiple?: boolean;
   subRowSelection?: boolean;
+  preSelectedRows?: string[];
```
Comment (yohernandez) [inline stencil-workspace/storybook/stories/components/modus-table/modus-table-storybook-docs.mdx:607]: Noticed that this property was missed in a previous PR

File: stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx [hunk 18]
```diff
@@ -12,14 +12,20 @@ export const ModusTableHeaderCheckbox: FunctionalComponent<ModusTableHeaderCheck
   const {
     tableInstance: { getIsAllRowsSelected, getIsSomeRowsSelected, getToggleAllRowsSelectedHandler },
     rowSelectionOptions,
+    density
   } = context;
+  let checkboxSize: 'medium' | 'small' = 'medium';
```
Comment (yohernandez) [inline stencil-workspace/src/components/modus-table/parts/row/selection/modus-table-header-checkbox.tsx:17]: Same for the header.

**Comment by yohernandez** [comment]
@cjwinsor The PR is ready for review. The property checkboxSize was removed. The checkbox size will change to small for compact density. The footer changes were removed too.

**Comment by dnarey** [comment]
@cjwinsor Resolved conflicts with modus-table.e2e.tx.  PR is ready for review.

**Review by cjwinsor — APPROVED** [review]

**Pull request merged**